messages/messages.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package messages provides functions for validating payloads from GitHub Webhooks.
+// GitHub docs: https://developer.github.com/webhooks/securing/#validating-payloads-from-github
+package messages
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"hash"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const (
+	// sha1Prefix is the prefix used by Github before the HMAC hexdigest.
+	sha1Prefix = "sha1="
+	// sha256Prefix and sha512Prefix are provided for future compatibility.
+	sha256Prefix = "sha256="
+	sha512Prefix = "sha512="
+	// eventHeader is the Github header key used to pass the event type.
+	eventHeader = "X-Github-Event"
+	// signatureHeader is the Github header key used to pass the HMAC hexdigest.
+	signatureHeader = "X-Hub-Signature"
+	// deliveryHeader is the Gihub header key used to pass the globally-unique Github Event ID.
+	deliveryHeader = "X-GitHub-Delivery"
+)
+
+// GenSignature generates a signature for the given payload.
+// hashType possible values are "sha1", "sha256", "sha512".
+// If hashType is empty or invalid, "sha1" is used.
+func GenSignature(payload, secretKey []byte, hashType string) string {
```

Comment:
Do we really need this function?  I can't think of a case where a user of the go-github library should ever need to generate a signature.  I know that you're using it in tests below, but I actually think we should use hardcoded, known signature values for tests.